tests/flow/test_query_mem_limit.py

```diff
@@ -30,45 +29,50 @@
                          WHERE (x / 2) = 50
                          RETURN x, count(x)"""
 
+def issue_query(conn, q, should_fail):
+    try:
+        g = Graph(GRAPH_NAME, conn)
+        g.query(q)
+        return not should_fail
+    except Exception as e:
+        assert "Query's mem consumption exceeded capacity" in str(e)
+        return should_fail
+
 class testQueryMemoryLimit():
     def __init__(self):
-        global g
+        # skip test if we're running under Valgrind
+        if Env().envRunner.debugger is not None:
+            Env().skip() # valgrind is not working correctly with replication
+
         self.env = Env(decodeResponses=True)
         self.conn = self.env.getConnection()
-        g = Graph(GRAPH_NAME, self.conn)
-
-    def issue_query(self, conn, q, should_fail):
-        try:
-            g.query(q)
-            self.env.assertFalse(should_fail)
-        except Exception as e:
-            assert "Query's mem consumption exceeded capacity" in str(e)
-            self.env.assertTrue(should_fail)
 
     def stress_server(self, queries):
-        threads = []
         connections = []
-        threadpool_size = self.conn.execute_command("GRAPH.CONFIG", "GET", "THREAD_COUNT")[1]
+        qs = []
+        should_fails = []
+        thread_count = self.conn.execute_command("GRAPH.CONFIG", "GET", "THREAD_COUNT")[1]
+        pool = Pool(nodes=thread_count)
 
         # init connections
-        for t in range(threadpool_size):
+        for t in range(thread_count):
             connections.append(self.env.getConnection())
 
-        # init circular iterator
-        connections_pool = cycle(connections)
+        for q in queries:
+            qs.append(q[0])
+            should_fails.append(q[1])
 
         # invoke queries
-        for q in queries:
-            con = next(connections_pool)
-            t = threading.Thread(target=self.issue_query, args=(con, q[0], q[1]))
-            t.setDaemon(True)
-            threads.append(t)
-            t.start()
-
-        # wait for threads to return
-        while len(threads) > 0:
-            t = threads.pop()
-            t.join()
+        m = pool.amap(issue_query, connections, qs, should_fails)
```

Comment:
Are we sure `len(connections) == len(qs)` ? if so can we make the construction more obvious ?

Reply:
yes you see that the for loop build them